packages/orchid/src/__tests__/example-task.ts

```diff
@@ -1,79 +1,53 @@
-import { TaskSpec, TaskContext, SpecToTask } from '../task.types';
-
-declare global {
-  interface GlobalTasks {
-    main: SpecToTask<typeof mainTask>;
-    double: SpecToTask<typeof doubleTask>;
-    list: SpecToTask<typeof listTask>;
-    addOne: SpecToTask<typeof addOneTask>;
-    sum: SpecToTask<typeof sumTask>;
-  }
-}
+import { TaskSpec } from '../task.types';
 
 export const mainTask: TaskSpec<undefined, Promise<number>> = {
-  name: 'main',
-  async run(input: undefined, { run, log }: TaskContext) {
-    const list = await run('list', 4);
+  id: 'main',
+  async run(input: undefined, { run, log }) {
+    const list = run(listTask, 4);
     log.info('Successful List', list);
     return list;
   },
 };
 
-export type DoubleTask = SpecToTask<typeof doubleTask>;
-export type ListTask = SpecToTask<typeof listTask>;
-
-export const doubleTask: TaskSpec<number, Promise<number>, number[]> = {
-  name: 'double',
-  async run(input, ctx: TaskContext<number[]>) {
-    // console.log('Task.run:double', input, 'parent', ctx.parent.name);
-    if (ctx.parent.name !== 'list') {
-      throw new Error('Double can only be called from list');
-    }
-    const listLength = ctx.parent.input.length;
-    if (listLength > 6) {
-      ctx.log.warn('log', 'List is too long, not doubling');
-      return input;
-    }
-    return input * 2;
-  },
-};
-
 export const listTask: TaskSpec<number, Promise<number>> = {
-  name: 'list',
-  async run(amount: number, ctx: TaskContext) {
-    // console.log('Task.run List', amount, 'parent', ctx.parent.name);
+  id: 'list',
+  async run(amount: number, { run }): Promise<number> {
     const list = new Array(amount).fill('').map((a, i) => i);
```

Comment:
Not important at all but my new favorite way to create arrays of a certain length without needing to fiddle around with holes in the array is `Array.from({ length: amount}).map(…)`. No need to `fill()` it in order to be able to map over it.

Reply:
Ahh I did use this api at some point, but then forgot again about it. 🙏